pkg/common/health/health.go

```diff
@@ -0,0 +1,95 @@
+package health
+
+import (
+	"context"
+	"net/http"
+	"sync"
+
+	"github.com/InVisionApp/go-health"
+	"github.com/InVisionApp/go-health/handlers"
+	"github.com/sirupsen/logrus"
+)
+
+// health.Checker is responsible for running health checks and serving the healthcheck HTTP paths
+type Checker struct {
+	config Config
+
+	server *http.Server
+
+	hc *health.Health
+
+	log logrus.FieldLogger
+}
+
+func hello(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "text/plain")
+	w.WriteHeader(http.StatusOK)
+}
+
+func NewChecker(config Config, log logrus.FieldLogger) *Checker {
+	hc := health.New()
+
+	var server http.Server
+	// Start HTTP server if address is configured
+	if config.Address != "" {
+		handler := http.NewServeMux()
+
+		handler.HandleFunc(config.getHealthPath(), handlers.NewJSONHandlerFunc(hc, nil))
+		handler.HandleFunc(config.getLivePath(), hello)
+
+		server = http.Server{
+			Addr:    config.Address,
+			Handler: handler,
+		}
+	}
+
+	hc.StatusListener = &statusListener{}
+
+	return &Checker{config: config, server: &server, hc: hc, log: log}
+}
+
+func (c *Checker) AddCheck(name string, checker health.ICheckable) error {
+	return c.hc.AddCheck(&health.Config{
```

Comment:
Quick review of go-health source leads me to believe that AddCheck isn't thread safe. Should we have a mutex on Checker struct?

Reply:
AddCheck can only be called before healthchecks are started, IE, it must be called during server startup.  I think we're single threaded there, but it's certainly safer to throw a mutex around it.